### ISSUE TITLE
Re-review DOM sync approach.

### ISSUE THREAD

**Issue by lifeart** [emberjs/ember.js#21039]
It seems we could explore different way to sync DOM with our state.
Instead of tags revalidation, we could explicitly mark "invalid" tags and resolve related opcodes to it.

```
Pseudocode:

TAG_ID = 12;

OpcodesForTag = {
  [TAG_ID] = [UpdateAttr, UpdateTextContent]
}

TagsToRevalidate = [];

Tag {
   id = TAG_ID;
   update() {
       this.value = 42;
       TagsToRevalidate.push(this);
   }
}

While (TRUE) {
   TagsToRevalidate.ForEach(Tag => {
      const value = tag.value;
      OpcodesForTag[this.id].forEach( opcode => {
          opcode(value);
      });
   });
}

```

Here is sample implementation concept: https://codepen.io/lifeart/pen/abMzEZm?editors=0110

Includes:

* helpers
* tags combination
* basic DOM update ops
* if condition

<img width="415" alt="image" src="https://github.com/emberjs/ember.js/assets/1360552/9380343d-68f7-48d8-b6c4-0ea4155498c8">

## Tldr:
current glimmer-vm tag invalidation approach:
 `dirtryTag -> scheduleRerender -> validate all existing tags in application -> iterate over all opcodes and if changed -> execute`
 approach in issue: 
`dirtyTag -> scheduleRerender -> getOpcodesForTag(tag) -> execute`
